public/app/features/logs/components/log-context/LogRowContextModal.tsx

```diff
@@ -0,0 +1,287 @@
+import { css } from '@emotion/css';
+import React, { useEffect, useLayoutEffect, useState } from 'react';
+import { useAsyncFn } from 'react-use';
+
+import {
+  DataQueryResponse,
+  DataSourceWithLogsContextSupport,
+  GrafanaTheme2,
+  LogRowContextOptions,
+  LogRowContextQueryDirection,
+  LogRowModel,
+  LogsDedupStrategy,
+  LogsSortOrder,
+  SelectableValue,
+} from '@grafana/data';
+import { config } from '@grafana/runtime';
+import { LoadingBar, Modal, useTheme2 } from '@grafana/ui';
+import { dataFrameToLogsModel } from 'app/core/logsModel';
+import store from 'app/core/store';
+import { SETTINGS_KEYS } from 'app/features/explore/utils/logs';
+
+import { LogRows } from '../LogRows';
+
+import { LoadMoreOptions, LogContextButtons } from './LogContextButtons';
+
+const getStyles = (theme: GrafanaTheme2) => {
+  return {
+    modal: css`
+      width: 85vw;
+      ${theme.breakpoints.down('md')} {
+        width: 100%;
+      }
+      top: 50%;
+      left: 50%;
+      transform: translate(-50%, -50%);
+    `,
+    entry: css`
+      position: sticky;
+      z-index: 1;
+      top: -1px;
+      bottom: -1px;
+      & > td {
+        padding: ${theme.spacing(1)} 0 ${theme.spacing(1)} 0;
+      }
+      background: ${theme.colors.emphasize(theme.colors.background.secondary)};
+
+      & > table {
+        margin-bottom: 0;
+      }
+    `,
+    datasourceUi: css`
+      padding-bottom: ${theme.spacing(1.25)};
+      display: flex;
+      align-items: center;
+    `,
+    logRowGroups: css`
+      overflow: auto;
+      max-height: 75%;
+      align-self: stretch;
+      display: inline-block;
+      & > table {
+        min-width: 100%;
+      }
+    `,
+    flexColumn: css`
+      display: flex;
+      flex-direction: column;
+    `,
+    flexRow: css`
+      display: flex;
+      flex-direction: row;
+      & > div:last-child {
+        margin-left: auto;
+      }
+    `,
+    noMarginBottom: css`
+      & > table {
+        margin-bottom: 0;
+      }
+    `,
+    hidden: css`
+      display: none;
+    `,
+  };
+};
+
+export enum LogGroupPosition {
+  Bottom = 'bottom',
+  Top = 'top',
+}
+
+interface LogRowContextModalProps {
+  row: LogRowModel;
+  open: boolean;
+  timeZone: string;
+  onClose: () => void;
+  getRowContext: (row: LogRowModel, options?: LogRowContextOptions) => Promise<DataQueryResponse>;
+  logsSortOrder?: LogsSortOrder | null;
+  runContextQuery?: () => void;
+  getLogRowContextUi?: DataSourceWithLogsContextSupport['getLogRowContextUi'];
+}
+
+export const LogRowContextModal: React.FunctionComponent<LogRowContextModalProps> = ({
+  row,
+  open,
+  logsSortOrder,
+  getLogRowContextUi,
+  onClose,
+  getRowContext,
+  timeZone,
+}) => {
+  const scrollElement = React.createRef<HTMLDivElement>();
+  const entryElement = React.createRef<HTMLTableRowElement>();
+
+  const theme = useTheme2();
+  const { modal, flexColumn, flexRow, entry, datasourceUi, logRowGroups, noMarginBottom, hidden } = getStyles(theme);
```

Comment:
Personal opinion, but I think destructuring styles as variables is potentially confusing. E.g. `hidden` looks like a boolean variable, not the classname "hidden". `styles.hidden` is longer but would make it less confusing.

Reply:
I like that. Changed in d0c24b065498f732a07531c3ba1babf7813c0447